sim/hunter/item_sets_pve.go

```diff
@@ -191,3 +191,170 @@ var ItemSetGiantstalkerPursuit = core.NewItemSet(core.ItemSet{
 		},
 	},
 })
+
+var ItemSetDragonstalkerProwess = core.NewItemSet(core.ItemSet{
+	Name: "Dragonstalker's Prowess",
+	Bonuses: map[int32]core.ApplyEffect{
+		// Raptor Strike increases the damage done by your next other melee ability within 5 sec by 20%.
+		2: func(agent core.Agent) {
+			hunter := agent.(HunterAgent).GetHunter()
+
+			procAura := hunter.RegisterAura(core.Aura{
+				ActionID: core.ActionID{SpellID: 467329},
+				Label:    "S03 - Item - T2 - Hunter - Melee 2P Bonus",
+				Duration: time.Second * 5,
+				OnGain: func(aura *core.Aura, sim *core.Simulation) {
+					for _, spell := range hunter.MeleeSpells {
+						if spell != nil {
```

Comment:
You should be safe to not check for nil here because you use `OnSpellRegistered` to fill this